src/libs/antares/study/parts/hydro/hydroreservoirlevels.cpp

```diff
@@ -0,0 +1,176 @@
+    Buffer.reset(3L, DAYS_PER_YEAR, true);
+    Buffer.fillColumn(ReservoirLevels::maximum, 1.);
+    Buffer.fillColumn(ReservoirLevels::average, 0.5);
+}
+
+bool ReservoirLevels::forceReload(bool reload) const
+{
+    bool ret = true;
+    ret = max.forceReload(reload) && ret;
+    ret = min.forceReload(reload) && ret;
+    ret = avg.forceReload(reload) && ret;
+    ret = Buffer.forceReload(reload) && ret;
+
+    return ret;
+}
+
+void ReservoirLevels::markAsModified() const
+{
+    max.markAsModified();
+    min.markAsModified();
+    avg.markAsModified();
+    Buffer.markAsModified();
+}
+
+bool ReservoirLevels::loadScenarizedReservoirLevels(const fs::path& folder)
+{
+    Matrix<>::BufferType fileContent;
+
+    bool ret = true;
+
+    fs::path filePath = folder / "maxDailyReservoirLevels.txt";
+    ret = max.timeSeries.loadFromCSVFile(filePath.string(), 1, DAYS_PER_YEAR, &fileContent) && ret;
+    filePath = folder / "minDailyReservoirLevels.txt";
+    ret = min.timeSeries.loadFromCSVFile(filePath.string(), 1, DAYS_PER_YEAR, &fileContent) && ret;
+    filePath = folder / "avgDailyReservoirLevels.txt";
+    ret = avg.timeSeries.loadFromCSVFile(filePath.string(), 1, DAYS_PER_YEAR, &fileContent) && ret;
+
+    return ret;
+}
+
+bool ReservoirLevels::loadReservoirLevels(const std::string& areaID,
+                                          const std::filesystem::path& folder,
+                                          bool usedBySolver,
+                                          bool useScenarizedResevoirLevels)
+{
+    bool ret = true;
+    Matrix<>::BufferType fileContent;
+
+    fs::path filePath = folder / "common" / "capacity"
+                        / std::string("reservoir_" + areaID + ".txt");
+
+    Buffer.reset(3, DAYS_PER_YEAR, true);
+
+    if (!usedBySolver)
+    {
+        bool enabledModeIsChanged = false;
+        if (JIT::enabled)
+        {
+            JIT::enabled = false;
+            enabledModeIsChanged = true;
+        }
+
+        ret = Buffer.loadFromCSVFile(filePath.string(),
+                                     3,
+                                     DAYS_PER_YEAR,
+                                     Matrix<>::optFixedSize,
+                                     &fileContent)
+              && ret;
+
+        if (enabledModeIsChanged)
+        {
+            JIT::enabled = true; // Back to the previous loading mode.
+        }
+    }
+    else if (!useScenarizedResevoirLevels)
+    {
+        ret = Buffer.loadFromCSVFile(filePath.string(),
+                                     3,
+                                     DAYS_PER_YEAR,
+                                     Matrix<>::optFixedSize,
+                                     &fileContent)
+              && ret;
+
+        copyReservoirLevelsFromBuffer();
+    }
+    else
+    {
+        filePath.clear();
+        filePath = folder / "series" / areaID;
+        ret = loadScenarizedReservoirLevels(filePath) && ret;
+    }
+
+    return ret;
+}
```

Comment:
Instead of this if..else, can you add 2 classes
- One responsible for loading in the existing format
- One responsible for loading data in the new format